pkg/pipeline/source/sdk.go

```diff
@@ -181,40 +182,49 @@ func (s *SDKSource) joinRoom() error {
 		if s.VideoEnabled {
 			tracks[s.VideoTrackID] = struct{}{}
 		}
-		err = s.awaitTracks(tracks)
+		w, h, err = s.awaitTracks(tracks)
 
 	case types.RequestTypeTrack:
 		fileIdentifier = s.TrackID
-		err = s.awaitTracks(map[string]struct{}{s.TrackID: {}})
+		w, h, err = s.awaitTracks(map[string]struct{}{s.TrackID: {}})
 	}
 	if err != nil {
 		return err
 	}
 
-	if err = s.UpdateInfoFromSDK(fileIdentifier, s.filenameReplacements); err != nil {
+	if err = s.UpdateInfoFromSDK(fileIdentifier, s.filenameReplacements, w, h); err != nil {
 		logger.Errorw("could not update file params", err)
 		return err
 	}
 
 	return nil
 }
 
-func (s *SDKSource) awaitParticipant(identity string) error {
+func (s *SDKSource) awaitParticipant(identity string) (uint32, uint32, error) {
 	s.errors = make(chan error, 2)
 
 	rp, err := s.getParticipant(identity)
 	if err != nil {
-		return err
+		return 0, 0, err
 	}
 
 	for trackCount := 0; trackCount == 0 || trackCount < len(rp.Tracks()); trackCount++ {
 		if err = <-s.errors; err != nil {
-			return err
+			return 0, 0, err
+		}
+	}
+
+	var w, h uint32
+	for _, t := range rp.Tracks() {
```

Comment:
what happens if the participant publishes video late?

Reply:
Currently, it will default to the video encoding settings (which themselves default to 1080p). I just added a small fix to make sure this is true.

Changing the behavior to start the image output only when a video track joins would be better, but I do not have any way to add the image output on the running pipeline yet. This will come when I implement "UpdateOutputs".